File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
    public:
        bool isAnagram(string s, string t) {
            unordered_map<char, int> count;
            for(char c : s){
                count[c - 'a']++;
            }
            for(char c : t){
                count[c - 'a']--;
            }

            for(auto it = count.begin(); it != count.end(); it++){
                if(it->second != 0) return false;
            }
            return true;
        }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 * this.val = (val===undefined ? 0 : val)
 * this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {void} Do not return anything, modify head in-place instead.
 */
class Solution {
    reorderList(head) {
        if (!head || !head.next || !head.next.next) return;

        let slow = head;
        let fast = head;
        while (fast && fast.next) {
            slow = slow.next;
            fast = fast.next.next;
        }

        let prev = null;
        let curr = slow.next;
        slow.next = null;
        while (curr) {
            let tempNext = curr.next;
            curr.next = prev;
            prev = curr;
            curr = tempNext;
        }

        let first = head;
        let second = prev;

        while (second) {
            let temp1 = first.next;
            let temp2 = second.next;

            first.next = second;
            second.next = temp1;

            first = temp1;
            second = temp2;
        }
    }
}